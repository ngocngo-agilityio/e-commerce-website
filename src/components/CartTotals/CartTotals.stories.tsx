import type { Meta, StoryObj } from '@storybook/react';

// Components
import CartTotals from '.';

const meta = {
  title: 'CartTotals',
  component: CartTotals,
  parameters: {
    layout: 'centered',
    nextjs: {
      appDirectory: true,
    },
  },
  tags: ['autodocs'],
} satisfies Meta<typeof CartTotals>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    total: 23.43,
    onCheckout: () => {},
  },
};
