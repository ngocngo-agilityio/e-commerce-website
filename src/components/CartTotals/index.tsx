// Libs
import { memo } from 'react';
import { Box, Flex, Heading, Text, Divider, Button } from '@chakra-ui/react';

// Utils
import { formatCurrency } from '@utils';

interface ICartTotalsProps {
  total: number;
  onCheckout: () => void;
}

const CartTotals = ({ total, onCheckout }: ICartTotalsProps): JSX.Element => {
  const formattedTotal = formatCurrency(total);

  return (
    <>
      <Heading mt="83px" mb="38px" fontSize={{ base: '3xl', md: '4xl' }}>
        Cart Totals
      </Heading>
      <Box w={{ base: 'full', md: '531px' }} mb="41px">
        <Flex justifyContent="space-between" py="17px">
          <Text>Subtotal</Text>
          <Text>{formattedTotal}</Text>
        </Flex>
        <Divider />
        <Flex justifyContent="space-between" py="17px">
          <Text>ShippingFree</Text>
          <Text>FREE!!!</Text>
        </Flex>
        <Divider />
        <Flex justifyContent="space-between" py="17px">
          <Text fontFamily="arimo" fontWeight="bold">
            Total
          </Text>
          <Text>{formattedTotal}</Text>
        </Flex>
      </Box>

      <Button
        variant="checkoutBtn"
        w={{ base: '200px', md: '262px' }}
        h={{ base: '40px', md: '58px' }}
        fontSize={{ base: 'md', md: 'xl' }}
        onClick={onCheckout}
      >
        Proceed to Checkout
      </Button>
    </>
  );
};

export default memo(CartTotals);
