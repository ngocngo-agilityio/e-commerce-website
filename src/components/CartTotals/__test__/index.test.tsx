// Libs
import { render } from '@testing-library/react';

// Sections
import CartTotals from '..';

describe('CartTotals page', () => {
  it('Should render match with snapshot.', async () => {
    const { container } = render(
      <CartTotals total={12.3} onCheckout={jest.fn()} />,
    );

    expect(container).toMatchSnapshot();
  });
});
