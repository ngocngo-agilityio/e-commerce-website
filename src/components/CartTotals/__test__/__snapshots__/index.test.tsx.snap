// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartTotals page Should render match with snapshot. 1`] = `
<div>
  <h2
    class="chakra-heading css-1vq0cl1"
  >
    Cart Totals
  </h2>
  <div
    class="css-1ebl0dg"
  >
    <div
      class="css-19y425s"
    >
      <p
        class="chakra-text css-0"
      >
        Subtotal
      </p>
      <p
        class="chakra-text css-0"
      >
        $12.30
      </p>
    </div>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-1upb9tn"
    />
    <div
      class="css-19y425s"
    >
      <p
        class="chakra-text css-0"
      >
        ShippingFree
      </p>
      <p
        class="chakra-text css-0"
      >
        FREE!!!
      </p>
    </div>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-1upb9tn"
    />
    <div
      class="css-19y425s"
    >
      <p
        class="chakra-text css-19q2mny"
      >
        Total
      </p>
      <p
        class="chakra-text css-0"
      >
        $12.30
      </p>
    </div>
  </div>
  <button
    class="chakra-button css-1jbx1h7"
    type="button"
  >
    Proceed to Checkout
  </button>
</div>
`;
